benchmark/bench.py

```diff
@@ -0,0 +1,518 @@
+        "-r", "--runs", default=10, help="Number of runs to perform for each command"
+    )
+    parser.add_argument(
+        "-o", "--output_file_name", default="summary.md", help="Name of the output file"
+    )
+    parser.add_argument(
+        "-b", "--bucket", required=True, help="Name of the bucket in remote"
+    )
+    parser.add_argument(
+        "-l",
+        "--local-path",
+        help="specify a local path for temporary files to be loaded.",
+    )
+    parser.add_argument(
+        "-p",
+        "--prefix",
+        default="s5cmd-benchmarks-",
+        help="Key prefix to be used while uploading to a specified bucket",
+    )
+    parser.add_argument(
+        "-hf",
+        "--hyperfine-extra-flags",
+        help="hyperfine global extra flags. "
+        "Write in between quotation marks "
+        "and start with a space to avoid bugs.",
+    )
+    parser.add_argument(
+        "-sf",
+        "--s5cmd-extra-flags",
+        default="",
+        help="s5cmd global extra flags. "
+        "Write in between quotation marks "
+        "and start with a space to avoid bugs.",
+    )
+
+    args = parser.parse_args(argv)
+
+    cwd = os.getcwd()
+
+    local_dir, dst_path = create_bench_dir(args.bucket, args.prefix, args.local_path)
+    print("The created local&remote files will be deleted at the end of tests.")
+    print(
+        f"Hyperfine will execute s5cmd uploads {args.warmup} times to warmup, and {args.runs} times for measurements."
+    )
+    old_s5cmd, new_s5cmd = build_s5cmd_exec(args.s5cmd[0], args.s5cmd[1], local_dir)
+
+    scenarios = [
+        Scenario(
+            name="small files",
+            cwd=cwd,
+            dst_path=dst_path,
+            local_dir=local_dir,
+            file_size="1M",
+            file_count="10000",
+            s5cmd_args=args.s5cmd_extra_flags,
+            hyperfine_args={
+                "runs": args.runs,
+                "warmup": args.warmup,
+                "extra_flags": args.hyperfine_extra_flags,
+            },
+        ),
+        Scenario(
+            name="large file",
+            cwd=cwd,
+            dst_path=dst_path,
+            local_dir=local_dir,
+            file_size="10G",
+            file_count="1",
+            s5cmd_args=args.s5cmd_extra_flags,
+            hyperfine_args={
+                "runs": args.runs,
+                "warmup": args.warmup,
+                "extra_flags": args.hyperfine_extra_flags,
+            },
+        ),
+        Scenario(
+            name="very large file",
+            cwd=cwd,
+            dst_path=dst_path,
+            local_dir=local_dir,
+            file_size="300G",
+            file_count="1",
+            s5cmd_args=args.s5cmd_extra_flags,
+            hyperfine_args={
+                "runs": "1",
+                "warmup": "0",
+                "extra_flags": args.hyperfine_extra_flags,
+            },
+        ),
+    ]
+
+    init_bench_results(cwd, args.output_file_name, scenarios)
```

Comment:
Is it possible to create the summary files if the scenario runs are successful.
If there is an error empty summary files are created in the working directory.

Reply:
with commit [6540bbc](https://github.com/peak/s5cmd/pull/471/commits/6540bbc73501a742863853524eecc93a44c77bc1), it will create the summary files if at least one full run is successful.